src/Modal/Modal.component.tsx

```diff
@@ -18,73 +28,82 @@ import { ModalClose } from './Close';
 
 export type ModalSize = 'lg' | 'sm';
 
-const Sizes = {
-    lg: { width: 50, contentPadding: 2.5, footerHeight: 6.5 },
-    sm: { width: 25, contentPadding: 2, footerHeight: 5.5 },
-};
-
-const defaultSize: ModalSize = 'lg';
+const MODAL_THEME_KEY = 'modal';
+
+const size = variant({
+    themeKey: MODAL_THEME_KEY,
+    prop: 'size',
+    variants: {
+        lg: {
+            width: 50,
+            contentPadding: 2.5,
+            footerHeight: 6.5,
+            height: 42.375,
+        },
+        sm: {
+            width: 25,
+            contentPadding: 2,
+            footerHeight: 5.5,
+            height: 42.375,
+        },
+    },
+    defaultValue: 'lg',
+});
 
 const StyledModal = StyledReactModal.default.styled`
     position: relative;
-    width: ${({ width, size = defaultSize }: ModalProps) =>
-        width || `${Sizes[size].width}rem`};
-    height: ${({ height = '42.375rem' }: ModalProps) => height};
-    margin: ${({ margin = 'auto' }: ModalProps) => margin};
+
+    ${compose(
+        marginStyles,
+        backgroundStyles,
+        colorStyles,
+        borderRadiusStyles
+    )}
+
+    ${size(
```

Comment:
This is using a custom `variant` util I made for handling more complex logic of theme settings.

The provided `variant` function from styled-components and styled-system would require each variant to be css style objects, which can't really handle this use-case.

Reply:
Alternatively I could do something like create a util for mapping the active variant properties as props to the component. This could potentially some prop name conflicts.